crates/runtime/src/dataconnector.rs

```diff
@@ -338,66 +340,60 @@ pub trait ListingTableConnector: DataConnector {
         )
     }
 
-    fn construct_metadata_provider(
-        &self,
-        dataset: &Dataset,
-    ) -> DataConnectorResult<Arc<dyn TableProvider>>
+    fn get_object_store(&self, dataset: &Dataset) -> DataConnectorResult<Arc<dyn ObjectStore>>
     where
         Self: Display,
     {
-        let ctx = Self::get_session_context();
         let store_url = self.get_object_store_url(dataset)?;
-
-        // TODO: For some reason, this `store` is not working. It's not setting `bucket_endpoint`.
-        // Need this, store-specific, prefix parsing
-        let (_store, prefix) = object_store::parse_url(&store_url.clone())
-            .boxed()
-            .context(UnableToConnectInternalSnafu {
-                dataconnector: format!("{self}"),
-            })?;
-        let (_, extension) = self.get_file_format_and_extension(dataset)?;
-
-        let (prefix_str, filename_regex) = if let Some(_ext) = prefix.extension() {
-            // Prefix is not collection, but a single file
-            let filename = prefix.filename().unwrap_or_default();
-            (
-                prefix
-                    .to_string()
-                    .strip_suffix(filename)
-                    .unwrap_or_default()
-                    .to_string(),
-                filename.to_string(),
-            )
-        } else {
-            (prefix.to_string(), format!(r"^.*\{extension}$"))
-        };
-
         let listing_store_url = ListingTableUrl::parse(store_url.clone()).boxed().context(
             UnableToConnectInternalSnafu {
                 dataconnector: format!("{self}"),
             },
         )?;
-        let object_store = ctx
+        Self::get_session_context()
             .runtime_env()
             .object_store(&listing_store_url)
             .boxed()
             .context(UnableToConnectInternalSnafu {
                 dataconnector: format!("{self}"),
-            })?;
+            })
+    }
 
-        let table =
-            ObjectStoreMetadataTable::try_new(object_store, Some(prefix_str), Some(filename_regex))
-                .context(InternalSnafu {
-                    dataconnector: format!("{self}"),
-                    code: "LTC-MP-OSMT".to_string(), // ListingTableConnector-MetadataProvider-ObjectStoreMetadataTableTryNew
-                })?;
+    fn construct_metadata_provider(
+        &self,
+        dataset: &Dataset,
+    ) -> DataConnectorResult<Arc<dyn TableProvider>>
+    where
+        Self: Display,
+    {
+        let store_url: Url = self.get_object_store_url(dataset)?;
+        let store = self.get_object_store(dataset)?;
+        let (_, extension) = self.get_file_format_and_extension(dataset)?;
+
+        let table = ObjectStoreMetadataTable::try_new(store, &store_url, Some(extension)).context(
```

Comment:
Let's also add a comment (similar to the others) explaining why this can't error.

Reply:
This can error?